code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -76,15 +76,6 @@ Please contact me on #coderbus IRC. ~Carnie x
 
 //HAIR OVERLAY
 /mob/living/carbon/human/update_hair()
-	//Reset our hair
-	remove_overlay(HAIR_LAYER)
-
-	if( (disabilities & HUSK) || (head && (head.flags & BLOCKHAIR)) || (wear_mask && (wear_mask.flags & BLOCKHAIR)) )
-		return

```

Comment:
I'm moving the head and mask check inside species' handle_hair() because it is sometimes called directly.